### ISSUE TITLE
Avoid suggesting ENV.fetch("FOO", nil) as replacement for ENV["FOO"]

### ISSUE THREAD

**Issue by wafcio** [rubocop/rubocop#14165]
## Is your feature request related to a problem? Please describe.

I'm frustrated with how the `Style/FetchEnvVar` cop works. It currently encourages replacing `ENV["FOO"]` with `ENV.fetch("FOO", nil)`, but this replacement doesn’t offer any real advantage. Using `fetch` with `nil` as the fallback essentially mimics the behavior of `ENV["FOO"]`, while losing readability and not leveraging the main benefit of `fetch`, which is to explicitly handle missing keys.

## Describe the solution you'd like

I’d like to propose that the cop be smarter about when it suggests using `ENV.fetch`. Specifically, it should avoid autocorrecting or flagging `ENV["FOO"]` usages unless a meaningful default value is being provided (i.e., something other than `nil`). If the intention is to allow for a fallback value, `fetch` makes sense; otherwise, it does not add value and may even reduce code clarity.

## Describe alternatives you've considered

One alternative is to disable the cop entirely, but that removes any enforcement for meaningful uses of `ENV.fetch`. Another idea is to allow configuration of acceptable default values or provide an option to ignore `nil` as a second argument.

## Additional context

The current autocorrect suggestion creates code like `ENV.fetch("FOO", nil)` which is functionally identical to `ENV["FOO"]`, but more verbose and arguably harder to read. This reduces developer confidence in the value provided by the cop.

**Comment by Yuhi-Sato** [comment]
@wafcio , @bbatsov 

Hi! I've reviewed this issue and implemented a solution to address the concerns raised.

## Proposed Solution

I've added a new configuration option `DefaultToNil` to the `Style/FetchEnvVar` cop that allows users to control the autocorrection behavior:

### `DefaultToNil: true` (default, preserves current behavior)
```ruby
# bad
ENV['X']

# good  
ENV.fetch('X', nil)
```

### `DefaultToNil: false` (new option)
```ruby
# bad
ENV['X']

# good
ENV.fetch('X')  # raises KeyError when missing
```


## Benefits of This Approach

- **Backward Compatibility**: Default behavior remains unchanged
- **Flexibility**: Users can choose the behavior that matches their intent
- **Clarity**: When `DefaultToNil: false`, missing environment variables result in clear errors
- **Readability**: Generates more appropriate code based on user intention

## Current Status

The implementation is nearly complete and all tests are passing. **Would it be acceptable to create a Pull Request** with these changes?

Please let me know if you have any questions or suggestions for improvements!

**Comment by Yuhi-Sato** [comment]
Hi @steiley

This is an idea I proposed, and as I mentioned in the issue, the implementation is almost complete, so I’ll submit a PR for it.
